lib/Rex/Logger.pm

```diff
@@ -264,6 +264,23 @@ sub format_string {
   return $line;
 }
 
+sub masq {
+  my ( $format, @params ) = @_;
+
+  return $format if scalar @params == 0;
+  return $format if scalar( grep { defined } @params ) == 0;
+
+  if ( exists $ENV{REX_DEBUG_INSECURE} && $ENV{REX_DEBUG_INSECURE} eq "1" ) {
+    return sprintf( $format, @params );
+  }
+
+  for my $p (@params) {
+    $p =~ s/./*/g;
```

Comment:
We might also just replace it with a fixed length string to avoid giving hints about the length of the masked sensitive data.